### DISCUSSION TITLE
Clear cache without instantly triggering reloads

### DISCUSSION THREAD

**Discussion by katywings** [solidjs/solid-router#470] [Ideas]
Currently the API only has a method to instantly `revalidate` caches. In some cases it would be useful to be able to just clear the cache without triggering the reload. Thinking of this situation: "You know that some data in the cache is old/invalid, but you don't already need the new data".

**Comment by ryansolid** [comment]
Yeah the APIs to do this are in there but I guess they aren't exposed. Mostly because of the short life of the cache on forward navigation. I was thinking we wouldn't need to invalidate anything manually that we weren't also refetching since odds are by the time you do the next thing, things would have changed.

I guess when I changed the behavior I think in 0.13.x to persist as long as the data is already on the page changed that assumption. Because now we don't fetch it again if it is already on the page, and that is something invalidating would allow for.

**Reply by katywings** [reply]
The thing with `revalidate` is, technically speaking you can already use it for soft-invalidation, if you make sure that nothing is depending on that cache, but sometimes its difficult to get the right timing. `invalidate` would be a bit simpler in such cases.

An example where I had to be nifty with revalidate:
https://codeberg.org/nitropage/nitropage/src/commit/a58f85d7abae870ac54ec4a270d3ab44c837f4c7/packages/nitropage/src/data/page.ts#L24

**Reply by mattbdc** [reply]
Please give us access to cacheKeyOp and then we can check the cacheEntry.mtime to trigger revalidation ourselves (possibly via a simple timeout X seconds later, or whenever we want).

I'd also like to know if a returned item is stale, again possible with cacheKeyOp if exposed.

**Reply by mattbdc** [reply]
```typescript
  const searchResults = createAsync(async () => {
      const response = fetchSearchResults(currentFilter())
      const isFetchedFromCacheHolder = { value: false };
      response.then(() => {
          isFetchedFromCacheHolder.value = true
      });
      await new Promise<void>((resolve) => setTimeout(() => resolve(), 0))
      const isFetchedFromCache = isFetchedFromCacheHolder.value
      const result = await response;
      await Object.defineProperty(result, 'isStale', {
          writable: true, enumerable: false, configurable: true, 
          value: isFetchedFromCache
      })
      return result;
  });
  ```
  
  nasty at moment
